apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Data;
+using System.Data.Common;
+
+// ReSharper disable UnusedParameter.Local
+
+namespace Google.Cloud.Spanner
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class SpannerParameter : DbParameter
+#if NET451
+        , ICloneable
+#endif
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        public SpannerParameter()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        public SpannerParameter(string spannerColumnName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        /// <param name="sourceDataTableName"></param>
+        public SpannerParameter(string spannerColumnName, string sourceDataTableName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        /// <param name="type"></param>
+        public SpannerParameter(string spannerColumnName, SpannerDbType type) : this(spannerColumnName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        /// <param name="type"></param>
+        public SpannerParameter(string spannerColumnName, DbType type) : this(spannerColumnName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        /// <param name="value"></param>
+        public SpannerParameter(string spannerColumnName, object value) : this(spannerColumnName)
```

Comment:
I'm tempted to remove this unless we definitely need it. I'd personally prefer everyone to be very specific about the type of parameter they're providing... but I know that does go against what already exists to some extent.